### PR TITLE
Improvement/zenko 4524 lint iam policies

### PR DESCRIPTION
**Pull request by KazToozs**
Linting of Iam policy files

### CODE REVIEW

**Review by anurag4DSB — APPROVED** [review]

File: tests/ctst/steps/iam-policies/IAMUser.ts [hunk 1]
```diff
@@ -1,42 +1,47 @@
...
-                    Effect: effect === 'Allow' ? 'Allow' : 'Deny',
-                    Action: `s3:${action}`,
-                    Resource: `arn:aws:s3:::${resource}`,
-                },
-            ],
-        })
-    });
-    this.saved.policyArn =
-        extractPropertyFromResults(await IAM.createPolicy(this.getCommandParameters()) as Result, 'Policy', 'Arn');
+// eslint-disable-next-line new-cap
```
Comment (ghost) [inline tests/ctst/steps/iam-policies/IAMUser.ts]: Considering that we use cucumber, this can go into the eslint configuration file directly, it will help in other projects

File: tests/ctst/steps/iam-policies/common.ts [hunk 2]
```diff
@@ -1,101 +1,107 @@
...
+When('the user tries to perform {string} on the bucket', async function (this: Zenko, action: string) {
+    let userCredentials: UserCredentials;
+    if ([EntityType.IAM_USER, EntityType.ACCOUNT].includes(this.getSaved().type as EntityType)) {
+        userCredentials = (this.parameters as { IAMSession: UserCredentials }).IAMSession;
         this.resumeRootOrIamUser();
     } else {
-        userCredentials = this.parameters.AssumedSession;
-        this.resumeAssumedRole();
+        userCredentials = (this.parameters as { AssumedSession: UserCredentials }).AssumedSession;
+        this.resumeRootOrIamUser();
```
Comment (ghost) [inline tests/ctst/steps/iam-policies/common.ts]: This looks unrelated to linting, is it something we need since recent changes?

File: tests/ctst/steps/iam-policies/common.ts [hunk 3]
```diff
@@ -1,101 +1,107 @@
...
+        assert.strictEqual((this.getResult() as {statusCode: number}).statusCode, 200);
+        break;
+    }
+    case 'GetAccountInfo': {
+        assert.strictEqual((this.getResult() as Error), false);
+        break;
+    }
+    default: {
+        assert.strictEqual((this.getResult() as {err: unknown}).err, null);
+    }
```
Comment (ghost) [inline tests/ctst/steps/iam-policies/common.ts]: the base function should create an object with a predictible type to avoid this kind of casting

File: tests/ctst/steps/iam-policies/common.ts (hunk 2 above)
Comment (KazToozs) [reply tests/ctst/steps/iam-policies/common.ts]: Indeed 9d2e9a7c80d7c6ed67a35678c85f45e803597d43

File: tests/ctst/world/Zenko.ts [hunk 4]
```diff
@@ -94,7 +99,7 @@ export interface ZenkoWorldParameters {
 export default class Zenko extends World<ZenkoWorldParameters> {
     private readonly command: string = '';
 
-    private result: object = {};
+    private result: ApiResult = {} as ApiResult;
```
Comment (ghost) [inline tests/ctst/world/Zenko.ts]: Not sure the `as` is needed here if you already create the variable of the right type, it should be properly ineferred by typescript

**Review by ghost — APPROVED** [review]

**Comment by KazToozs** [comment]
ping
